src/types.ts

```diff
@@ -1,13 +1,19 @@
+import { Logger as PinoLogger } from "pino"
 import { User } from "./schema"
 
 export type Side = "buy" | "sell"
 export type Currency = "USD" | "BTC"
 
 export type ISuccess = boolean
 
+export type Logger = PinoLogger
+
+export type Primitive = string | boolean | number
+
 export interface ILightningWalletUser {
-  user: any
-  logger: any
+  // FIXME: Add a type for user here.
```

Comment:
Adding a potential approach here for typing `User`: https://tomanagle.medium.com/strongly-typed-models-with-mongoose-and-typescript-7bc2f7197722

Reply:
Thanks for sharing that article @vindard. I remember we referred to it when thinking about typing mongoose a few months ago, but only now I noticed that the first section makes a reference to `typegoose`. They don't consider it because it 'adds too much abstraction'.

Can @bodymindarts and @samerbuna also look at that library and share their opinion on it, and whether we can consider using it?

Reply:
From reading the article my opinion is:
- using interfaces to add typing to the objects returned from the database 👍 
- using objects returned from the database as a place for domain functions 👎 

There is a big misunderstanding when talking about "the model". What model? Persistence model? Data model? Domain model? These are all distinct things and when things get complex need to be treated explicitly.

The current approach to add domain logic to the database entities is very common and okay-ish in the short term for small projects.... but long term it has a lot of negative consequences.

Anyway that's a topic to tackle in the future. For now adding any typing (for example via the interface approach mentioned in the article) would be a win!